src/Aspire.Hosting/MongoDB/MongoDbBuilderExtensions.cs

```diff
@@ -0,0 +1,92 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net.Sockets;
+using System.Text.Json;
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting.MongoDb;
+
+/// <summary>
+/// Provides extension methods for adding MongoDB resources to an <see cref="IDistributedApplicationBuilder"/>.
+/// </summary>
+public static class MongoDbBuilderExtensions
+{
+    private const int DefaultContainerPort = 27017;
+    private const string DefaultPassword = "password";
+    private const string PasswordEnvVarName = "MONGO_INITDB_ROOT_PASSWORD";
+
+    /// <summary>
+    /// Adds a MongoDB container to the application model. The default image is "mongo" and the tag is "latest".
+    /// </summary>
+    /// <returns></returns>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <param name="port">The host port for MongoDB.</param>
+    /// <param name="password">The password for the MongoDB root user. Defaults to a 'password' password.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{MongoDbContainerResource}"/>.</returns>
+    public static IResourceBuilder<MongoDbContainerResource> AddMongoDbContainer(
+        this IDistributedApplicationBuilder builder,
+        string name,
+        int port = DefaultContainerPort,
+        string password = DefaultPassword)
+    {
+        var mongoDbContainer = new MongoDbContainerResource(name, password);
+
+        return builder
+            .AddResource(mongoDbContainer)
+            .WithAnnotation(new ManifestPublishingCallbackAnnotation(WriteMongoDbContainerToManifest))
```

Comment:
We have a new method for this:

`.WithManifestPublishingCallback(...)`

Reply:
I've changed to use the new method, please review.